### PR TITLE
pmtiles: init at 1.12.0

### PR DESCRIPTION
**Pull request by Theaninova**
## Description of changes

Adds [`pmtiles`](https://github.com/protomaps/go-pmtiles/blob/main/main.go) at v1.12.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/gis/pmtiles/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  postInstall = ''
+    mv $out/bin/go-pmtiles $out/bin/pmtiles
+  '';
+
+  meta = with lib; {
+    description = "The single-file utility for creating and working with PMTiles archives.";
+    homepage = "https://github.com/protomaps/go-pmtiles";
+    license = licenses.bsd3;
+    maitainers = [ maintainers.theaninova ];
```
Comment (kirillrdy) [inline pkgs/applications/gis/pmtiles/default.nix:25]:
```suggestion
    maintainers = [ maintainers.theaninova ];
```

File: pkgs/by-name/pm/pmtiles/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  vendorHash = "sha256-gLFwGEUeH41bObG32MZznF7clct3h2GEvdZ2/KIiVb4=";
+
+  ldflags = [ "-s" "-w" "-X main.version=${version}" "-X main.commit=v${version}" ];
+
+  postInstall = ''
+    mv $out/bin/go-pmtiles $out/bin/pmtiles
+  '';
+
+  meta = with lib; {
+    description = "The single-file utility for creating and working with PMTiles archives.";
```
Comment (kirillrdy) [inline pkgs/by-name/pm/pmtiles/package.nix:22]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

please remove trailing coma

File: pkgs/by-name/pm/pmtiles/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
+  pname = "pmtiles";
+  version = "1.11.3";
```
Comment (kirillrdy) [inline pkgs/by-name/pm/pmtiles/package.nix:4]: optional: 1.12.0 is out, I'll leave it up to you to decide if you want to do part of this PR or subsequent PR

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -18338,6 +18338,15 @@
       fingerprint = "D2A2 F0A1 E7A8 5E6F B711  DEE5 63A4 4817 A52E AB7B";
     }];
   };
+  theaninova = {
```
Comment (kirillrdy) [inline maintainers/maintainer-list.nix:18341]: `a` should sorted be after `-`

Comment (Theaninova) [reply maintainers/maintainer-list.nix:18341]: Hm, that's how I had it at first but the "check maintainer list is sorted" check yelled at me that I should put it here.

Comment (kirillrdy) [reply maintainers/maintainer-list.nix:18341]: hmm ok, let me look into it, at least my editor and sorted - before a :-)

Comment (kirillrdy) [reply maintainers/maintainer-list.nix:18341]: ok, I've learned something new ! maintainers/scripts/check-maintainers-sorted.nix removes `-` and `_` so `thean` comes before `the-ar`

Comment (Theaninova) [reply maintainers/maintainer-list.nix:18341]: Huh, so I suppose that sorting is correct then?

Comment (kirillrdy) [reply maintainers/maintainer-list.nix:18341]: yes, sorting is correct

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 279791` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pmtiles</li>
  </ul>
</details>

thank you for your contribution and welcome to nixpkgs !

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**